internal/internal_versioning_client.go

```diff
@@ -0,0 +1,319 @@
+	BuildIDTaskReachabilityReachable
+	// BuildIDTaskReachabilityClosedWorkflowsOnly specifies that this Build ID does not have open workflows
+	// and is not reachable by new workflows, but MAY have closed workflows within the namespace retention period.
+	// Not applicable to activity-only task queues.
+	BuildIDTaskReachabilityClosedWorkflowsOnly
+	// BuildIDTaskReachabilityUnreachable indicates that this Build ID is not used for new executions, nor
+	// it has been used by any existing execution within the retention period.
+	BuildIDTaskReachabilityUnreachable
+)
+
+type (
+	// TaskQueueVersionSelection is a task queue filter based on versioning.
+	// It is an optional component of [DescribeTaskQueueEnhancedOptions].
+	TaskQueueVersionSelection struct {
+		// Include specific Build IDs.
+		BuildIDs []string
+		// Include the unversioned queue.
+		Unversioned bool
+		// Include all active versions. A version is active if it has had new
+		// tasks or polls recently.
+		AllActive bool
+	}
+
+	// DescribeTaskQueueEnhancedOptions is the input to [Client.DescribeTaskQueueEnhanced].
+	DescribeTaskQueueEnhancedOptions struct {
+		// Name of the task queue. Sticky queues are not supported.
+		TaskQueue string
+		// An optional queue selector based on versioning. If not provided,
+		// the result for the default Build ID will be returned. The default
+		// Build ID is the one mentioned in the first unconditional Assignment Rule.
+		// If there is no default Build ID, the result for the
+		// unversioned queue will be returned.
+		Versions *TaskQueueVersionSelection
+		// Task queue types to report info about. If not specified, all types are considered.
+		TaskQueueTypes []TaskQueueType
+		// Include list of pollers for requested task queue types and versions.
+		ReportPollers bool
+		// Include task reachability for the requested versions and all task types
+		// (task reachability is not reported per task type).
+		ReportTaskReachability bool
+	}
+
+	// WorkerVersionCapabilities includes a worker's build identifier
+	// and whether it is choosing to use the versioning feature.
+	// It is an optional component of [PollerInfo].
+	WorkerVersionCapabilities struct {
+		BuildID       string
+		UseVersioning bool
+	}
+
+	// PollerInfo provides information about a worker/client polling a task queue.
+	// It is used by [TaskQueueTypeInfo].
+	PollerInfo struct {
+		// Time of the last poll. A value of zero means it was not set.
+		LastAccessTime time.Time
+		// The identity of the worker/client who is polling this task queue.
+		Identity string
+		// Polling rate. A value of zero means it was not set.
+		RatePerSecond float64
+		// Optional poller versioning capabilities. Available when a worker has opted into the worker versioning feature.
+		WorkerVersionCapabilities *WorkerVersionCapabilities
+	}
+
+	// TaskQueueTypeInfo specifies task queue information per task type and Build ID.
+	// It is included in [TaskQueueVersionInfo].
+	TaskQueueTypeInfo struct {
```

Comment:
Is there a reason we did not include `TaskQueueStats`?

Reply:
They are coming, see this PR  https://github.com/temporalio/api/pull/414, but not waiting for this ...

Reply:
Yep  all good, for some reason I thought `TaskQueueStats` was going to be done the same time versioning was.